main.tf

```diff
@@ -98,6 +114,27 @@ resource "google_pubsub_subscription" "pull_subscriptions" {
   depends_on = [google_project_service.pubsub_api]
 }
 
+resource "google_pubsub_subscription" "bigquery_subscriptions" {
```

Comment:
Instead of creating new reousrces, can we leverage on the existing `pull_subscriptions` resources? And use the way how `dynamic "dead_letter_policy"` bock is used in the `push_subscriptions` ?

Reply:
Since "Write to BigQuery" is a separate subscription type in [GCP console](https://cloud.google.com/pubsub/docs/create-subscription#create_a_subscription), we will keep it as a separate resource.